pkg/bpf/map.go

```diff
@@ -747,21 +745,14 @@ func ConvertKeyValue(bKey []byte, bValue []byte, key interface{}, value interfac
 // MetadataDiff compares the metadata of the BPF maps and returns false if the
 // metadata does not match
 func (m *Map) MetadataDiff(other *Map) bool {
-	if m == nil || other == nil {
+	switch {
+	case m == other:
+		return true
+	case m == nil || other == nil:
 		return false
+	default:
+		return m.DeepEquals(other)
 	}
-
-	// create copies
-	m1 := *m
-	m2 := *other
-
-	// ignore fd in diff
-	m1.fd = 0
-	m2.fd = 0
-
-	logging.MultiLine(log.Debug, comparator.Compare(m1, m2))
```

Comment:
The debug diff is now gone

Reply:
Yes. Calling `comparator.Compare` with `Map` parameters makes `go tool vet` fail, because that creates a copy of each `Map` parameter, and hence a copy of their `Mutex` fields.
To fix this, I'll have to duplicate the logic of `comparator.compare` and use the `difflib` package directly.

Reply:
I'll do that shortly.

Reply:
Actually, passing `*Map` values instead of `Map` should work. I'll bring back the call to `Compare`, but directly with pointers. And I will still remove the first copies into `m1` and `m2` to avoid the `Mutex` copies.

Reply:
Done.